tests/sqlparser_common.rs

```diff
@@ -1806,11 +1806,34 @@ fn parse_join_nesting() {
         vec![join(nest!(nest!(nest!(table("b"), table("c")))))]
     );
 
-    let res = parse_sql_statements("SELECT * FROM (a NATURAL JOIN (b))");
+
+    let sql = "SELECT * FROM a NATURAL JOIN (((b NATURAL JOIN c)))";
+    let select = verified_only_select(sql);
+    let from = only(select.from);
+    assert_eq!(from.relation, table("a"));
     assert_eq!(
-        ParserError::ParserError("Expected joined table, found: )".to_string()),
-        res.unwrap_err()
+        from.joins,
+        vec![join(nest!(nest!(nest!(table("b"), table("c")))))]
+    );
+
+    let sql = "SELECT * FROM (a NATURAL JOIN (b))";
```

Comment:
It would be cleaner to have only the generic dialect accept this, and have a TODO test that makes the ANSI dialect fail it. But I guess that introducing the necessary infrastructure is too much for this PR.

So at least we must note that this test is supposed to succeed with the Snowflake dialect only:
```
// Parenthesized table names are non-standard, but supported in Snowflake SQL
```

Reply:
Agree that dialect  will be much cleaner , Sound like small project  :) . 
Added the comment.